File: hazelnet-cardano-shared/src/main/kotlin/io/hazelnet/shared/data/SharedWhitelist.kt
package io.hazelnet.shared.data

import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer

data class SharedWhitelist(
    @field:JsonSerialize(using = ToStringSerializer::class)
    val guildId: Long,
    val guildName: String,
    val whitelistName: String,
    val whitelistDisplayName: String,
    val signups: Set<WhitelistSignup>,
)

File: vibrantnet/core/src/types/whitelist/sharedWhitelist.ts
import { SharedWhitelistSignup } from './sharedWhitelistSignup';
import { WhitelistType } from './whitelistType';

export type SharedWhitelist = {
  guildId: string
  guildName: string
  whitelistName: string
  whitelistDisplayName: string
  type: WhitelistType
  signups: SharedWhitelistSignup[],
};
